File: app/src/main/java/com/example/miniproject/RetrofitService/ServiceClient.java
package com.example.miniproject.RetrofitService;

import com.example.miniproject.Model.AbstractResponse;
import com.example.miniproject.Model.ArticleModel;
import com.example.miniproject.Model.SourceModel;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface ServiceClient {
    String baseUrl = "https://newsapi.org";

    @GET("/v2/sources")
    Call<AbstractResponse> getSourcesList(/*@Query("apiKey") String apiKey*/);

    @GET("/v2/everything")
    Call<AbstractResponse> getArticlesList(@Query("sources") String sourceId/*, @Query("apiKey") String apiKey*/);
}

File: app/src/main/java/com/example/miniproject/retrofitService/ServiceClient.kt
package com.example.miniproject.retrofitService

import com.example.miniproject.model.AbstractResponse
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface ServiceClient {
    @get:GET("/v2/sources")
    val sourcesList: Call<AbstractResponse?>?

    @GET("/v2/everything")
    fun getArticlesList(@Query("sources") sourceId: String?): Call<AbstractResponse?>?

    companion object {
        const val baseUrl = "https://newsapi.org"
    }
}